backpack/core/derivatives/adaptive_avg_pool_nd.py

```diff
@@ -0,0 +1,112 @@
+
+
+class AdaptiveAvgPoolNDDerivatives(AvgPoolNDDerivatives):
+    """Implements the derivatives for AdaptiveAvgPool."""
+
+    def __init__(self, N: int):
+        """Initialization.
+
+        Args:
+            N: number of dimensions
+        """
+        super(AdaptiveAvgPoolNDDerivatives, self).__init__(N=N)
+
+    def check_parameters(
+        self, module: Union[AdaptiveAvgPool1d, AdaptiveAvgPool2d, AdaptiveAvgPool3d]
+    ) -> None:
+        """Checks if the parameters are supported.
+
+        Specifically checks if input shape is multiple of output shape.
+        In this case, there are parameters for AvgPoolND that are equivalent.
+
+        https://stackoverflow.com/questions/53841509/how-does-adaptive-pooling-in-pytorch-work/63603993#63603993 # noqa: B950
+
+        Args:
+            module: module to check
+
+        Raises:
+            NotImplementedError: if the given shapes do not match
+        """
+        shape_input: Size = module.input0.shape
+        shape_target: Size = module.output.shape
+
+        # check length of input shape
+        if not len(shape_input) == (self.N + 2):
+            raise NotImplementedError(
+                f"input must be (batch_size, C, ...) with ... {self.N} dimensions"
+            )
+        # check length of target shape
+        if not len(shape_target) == (self.N + 2):
+            raise NotImplementedError(
+                f"shape of target should have {self.N} dimensions"
+            )
+
+        # check if input shape is multiple of output shape
+        for n in range(self.N):
+            in_dim: int = shape_input[2 + n]
+            out_dim: int = shape_target[2 + n]
+            if (in_dim % out_dim) != 0:
+                raise NotImplementedError(
+                    "Not supported in BackPACK: Input shape of AdaptiveAvgPool "
+                    "must be multiple of output shape. Only in this case, there is "
+                    "an equivalent AvgPool module."
+                )
+
+    def get_parameters(
+        self, module: Union[AdaptiveAvgPool1d, AdaptiveAvgPool2d, AdaptiveAvgPool3d]
+    ) -> Tuple[List[int], List[int], List[int]]:
+        """Return parameters for an equivalent AvgPool.
+
+        Assumes that check_parameters has been run before.
+        Therefore, does not check parameters.
+
+        Args:
+            module: module to compute on
+
+        Returns:
+            stride, kernel_size, padding as lists of length self.N
+        """
+        shape_input: Size = module.input0.shape
+        shape_target: Size = module.output.shape
+
+        # calculate equivalent AvgPoolND parameters
+        stride: List[int] = []
+        kernel_size: List[int] = []
+        for n in range(self.N):
+            in_dim: int = shape_input[2 + n]
+            out_dim: int = shape_target[2 + n]
+            stride.append(in_dim // out_dim)
+            kernel_size.append(in_dim - (out_dim - 1) * stride[n])
+
```

Comment:
- Explicitly add `padding = [0 for _ in range(N)]`
- Adapt `return ..., ..., padding`

Reply:
Yes.